src/content/docs/en/guides/cms/umbraco.mdx

```diff
@@ -0,0 +1,250 @@
+
+For more information on Layouts see the [Astro documentation](https://docs.astro.build/en/basics/layouts/).
+
+Add the following markup to the `Layout.astro` file:
+
+```astro title="src/layouts/Layout.astro"
+---
+---
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>My Blog with Astro and Umbraco</title>
+</head>
+<body>
+    <main>
+        <slot />
+    </main>
+</body>
+</html>
+```
+
+We will use the homepage `index.astro` file at the root to list out the blog articles.
+
+For this we will use a `fetch` to call the Content Delivery API `content` endpoint and filter by contentType of 'article'.
+
+The article objects will include the properties and content set in the CMS.
+
+We can then loop through the articles and display a desired listing with a link to the article.
+
+Replace the default contents of `index.astro` with the below:
+
+```astro title="src/pages/index.astro"
+---
+import Layout from '../layouts/Layout.astro';
+const res = await fetch('http://localhost/umbraco/delivery/api/v2/content?filter=contentType:article');
+const articles = await res.json();
+---
+<Layout>
+	<h2>Blog Articles</h2>
+	{
+        articles.items.map((article: any) => (
+            <div>
+                <h3>{article.properties.title}</h3>
+                <p>{article.properties.articleDate}</p>
+                <a href={article.route.path}>Read more</a>
+            </div>
+        ))
+    }
+</Layout>
+```
+
+### Generating individual blog posts
+
+We will now create the file `[...slug].astro` file at the root of the `pages` directory.
+
+The `[...slug]` convention here will be used to create the Dynamic Routes for our articles. Read more about Dynamic Routing [here](https://docs.astro.build/en/guides/routing/#dynamic-routes)
+
+Here we use the `getStaticPaths()` function to return an an array of objects that represent out pages, in this case, our blog articles.
+
+The `slug` property on `params` is used to generate the URL path of the page. This is defined by the `[...slug].astro` naming convention.
+
+We are also passing a `props` object that will include the article returned from the API.
+
+These properties can then be used within our markup by defining `const { article } = Astro.props;`
+
+```astro title="[...slug].astro"
+---
+import Layout from '../layouts/Layout.astro';
+
+export async function getStaticPaths() {
+    let data = await fetch("http://localhost/umbraco/delivery/api/v2/content?filter=contentType:article");
+    let articles = await data.json();
+    
+    return articles.items.map((article: any) => ({
+        params: { slug: article.route.path },
+        props: { article: article },
+    }));
+}
+
+const { article } = Astro.props;
+---
+
+<Layout>
+  <h1>{article.properties.title}</h1>
+  <p>{article.properties.articleDate}</p>
+  <div set:html={article.properties.content?.markup}></div>
+</Layout>
+```
```

Comment:
``````suggestion
Create the file `[...slug].astro` at the root of the `/pages/` directory. This file will be used to [generate the individual blog pages dynamically](https://docs.astro.build/en/guides/routing/#dynamic-routes).

Note that the `params` property, which generates the URL path of the page, contains `article.route.path` from the API fetch. Similarly, the `props` property must include the entire `article` itself so that you can access all the information in your CMS entry.

Add the following code to `[...slug].astro` which will create your individual blog post pages:

```astro title="[...slug].astro"
---
import Layout from '../layouts/Layout.astro';

export async function getStaticPaths() {
    let data = await fetch("http://localhost/umbraco/delivery/api/v2/content?filter=contentType:article");
    let articles = await data.json();
    
    return articles.items.map((article: any) => ({
        params: { slug: article.route.path },
        props: { article: article },
    }));
}

const { article } = Astro.props;
---

<Layout>
  <h1>{article.properties.title}</h1>
  <p>{article.properties.articleDate}</p>
  <div set:html={article.properties.content?.markup}></div>
</Layout>
```

Your project should now contain the following files:

<FileTree>
- src/
  - layouts/
    - Layout.astro
  - pages/
    - index.astro
    - **[...slug].astro**
</FileTree>
``````
I shortened the description of how `getStaticPaths()` works because that's all available at the link to our own dynamic routing content. What *is* maybe worth pointing out is how/what to include from the fetch call to the Content Delivery API so that the information is properly available!